cpp/src/datetime/datetime_ops.cu

```diff
@@ -106,6 +107,28 @@ struct extract_last_day_of_month {
   }
 };
 
+// Number of days until month indexed by leap year and month (0-based index)
+static __device__ int16_t const days_until_month[2][12] = {
+  {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334},  // For non leap years
```

Comment:
Wow, this is really quite clever.

Reply:
Thanks! But to be honest I think this is a standard approach that I modified to work with CUDA.  Lookup tables are the oldest trick in the book.  :)

FWIW, @sriramch showed above that he has code that is 95% as fast without the lookup table, but it's blocked by an NVCC bug.